rosidl_parser/rosidl_parser/grammar.lark

```diff
@@ -75,13 +76,7 @@ string_literal: "\"\"" | "\"" /(\\\"|[^"])+/ "\""
 wide_string_literal: "L\"\"" | "L\"" /(\\\"|[^"])+/ "\""
 
 // 7.2.6.4 Floating-point Literals
-floating_pt_literal: DIGIT+ floating_pt_literal_dot DIGIT+ floating_pt_literal_e DIGIT*
-  | floating_pt_literal_dot DIGIT+ (floating_pt_literal_e DIGIT*)?
-  | DIGIT+ floating_pt_literal_dot DIGIT* (floating_pt_literal_e DIGIT*)?
-  | DIGIT+ floating_pt_literal_e DIGIT*
-// separate rules to identify the components
-floating_pt_literal_dot: "."
-floating_pt_literal_e: "e"i
+floating_pt_literal: FLOAT
 
 // 7.2.6.5 Fixed-Point Literals
 fixed_pt_literal: DIGIT+ "." DIGIT+ "d"i
```

Comment:
The definition of `fixed_pt_literal` seems incorrect since it doesn't allow the following part of 7.2.6.5:

> Either the integer part or the fraction part (but not
both) may be missing; the decimal point (but not the letter d or D ) may be missing.

Proposal: `fixed_pt_literal: DECIMAL "d"i`

Reply:
You are right, the fixed point literal also wasn't quite correct.  However, I ended up with the following grammar instead:

```
fixed_pt_literal: DECIMAL "d"i
  | INT "d"i
```

That's to take care of the case where no dot is in the number, as allowed by 7.2.6.5.  See 77f0373 for fixes.  I also added some more test cases to the floating_pt_literal.  I think this is now ready for review, so switching it to a real PR.